Platon/backend/app/search_engine/views.py

```diff
@@ -0,0 +1,149 @@
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                if related_job is not None:
+                    try:
+                        owner_list = User.query.filter(User.job_id == related_job.id).all()
+                    except: 
+                        return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                    for user in owner_list:
+                        id_list = [i[0] for i in result_id_score]
+                        if user.id not in id_list:
+                            result_list.append({"id":user.id, "name": user.name, "surname": user.surname, 
+                                            "profile_photo" : user.profile_photo, "is_private": user.is_private, "job_id" : user.job_id})
+                            result_id_score.append((user.id,score))
+                        else:
+                            id_index = id_list.index(user.id)
+                            result_id_score[id_index] =(user.id, result_id_score[id_index][1]+score) 
+            # Search tokens in Skills Table
+            for token,score in tokens:
+                try:
+                    related_skill = Skills.query.filter(func.lower(Skills.name) == token).first()
+                except: 
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                if related_skill is not None:
+                    try:
+                        owner_id_list = UserSkills.query.filter(UserSkills.skill_id == related_skill.id).all()
+                    except: 
+                        return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                    for owner_id in owner_id_list:
+                        id_list = [i[0] for i in result_id_score]
+                        if owner_id.user_id not in id_list:
+                            # Take user information from the DB
+                            try:
+                                user = User.query.filter(User.id == owner_id.user_id).first()
+                            except:
+                                return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                            result_list.append({"id":user.id, "name": user.name, "surname": user.surname, 
+                                            "profile_photo" : user.profile_photo, "is_private": user.is_private, "job_id" : user.job_id})
+                            result_id_score.append((user.id,score))
+                        else:
+                            id_index = id_list.index(owner_id.user_id)
+                            result_id_score[id_index] =(user.id, result_id_score[id_index][1]+score) 
+            # Search tokens for name, surname, institution or email match
+            for token,score in tokens:
+                try:
+                    query = '(LOWER(name) REGEXP ".*{0}.*" OR LOWER(surname) REGEXP ".*{0}.*" \
```

Comment:
please enlighten us about this REGEX operation

Reply:
It tries to find an occurance in the fields of the records. I thnik it will be much more explanatory if you search regular expression on the Internet.